controllers/fenceagentsremediation_controller_test.go

```diff
@@ -209,16 +211,77 @@ var _ = Describe("FAR Controller", func() {
+		conditionStatusPointer := func(status metav1.ConditionStatus) *metav1.ConditionStatus { return &status }
+		BeforeEach(func() {
+			farPod := createRunningPod("far-manager-test", farPodName, "")
+			DeferCleanup(deletePod, farPod)
+		})
+		JustBeforeEach(func() {
+			// Create node, and at the end clean it up with DeferCleanup
+			Expect(k8sClient.Create(context.Background(), node)).To(Succeed())
+			DeferCleanup(k8sClient.Delete, context.Background(), node)
+		})
+
+		When("creating valid FAR CR", func() {
+			var va *storagev1.VolumeAttachment
+			var terminatingPod *corev1.Pod
+			BeforeEach(func() {
+				node = utils.GetNode("", workerNode)
+				underTestFAR = getFenceAgentsRemediation(workerNode, fenceAgentIPMI, testShareParam, testNodeParam, v1alpha1.OutOfServiceTaintRemediationStrategy)
+				Expect(k8sClient.Create(context.Background(), underTestFAR)).To(Succeed())
+				va = createVA(vaName1, workerNode)
+				terminatingPod = createTerminatingPod("far-test-1", testPodName, workerNode)
+			})
+			// Create two VAs and two pods, and at the end clean them up with DeferCleanup
+			It("should have both far taint and out-of-service taint, and at the end they will be deleted", func() {
+				By("Searching for remediation taint")
+				Eventually(func(g Gomega) bool {
+					g.Expect(k8sClient.Get(context.Background(), nodeKey, node)).To(Succeed())
+					g.Expect(k8sClient.Get(context.Background(), farNamespacedName, underTestFAR)).To(Succeed())
+					res, _ := cliCommandsEquality(underTestFAR)
+					return utils.TaintExists(node.Spec.Taints, &farNoExecuteTaint) && utils.TaintExists(node.Spec.Taints, &outOfServiceTaint) && res
+				}, timeoutFinalizer, pollInterval).Should(BeTrue(), "both far taint and out-of-service taint should be added, and command format is correct")
+
+				// If the taints were added, then definitely the finalizer was added as well
+				By("Having a finalizer if we have a remediation taint")
+				Expect(controllerutil.ContainsFinalizer(underTestFAR, v1alpha1.FARFinalizer)).To(BeTrue())
+
+				// simulate the out-of-service taint by Pod GC Controller
+				deleteVA(va)
+				deletePod(terminatingPod)
+
+				By("Verifying correct conditions for successfull remediation")
+				Expect(underTestFAR.Status.LastUpdateTime).ToNot(BeNil())
+				verifyStatusCondition(workerNode, commonConditions.ProcessingType, conditionStatusPointer(metav1.ConditionFalse))
+				verifyStatusCondition(workerNode, v1alpha1.FenceAgentActionSucceededType, conditionStatusPointer(metav1.ConditionTrue))
+				verifyStatusCondition(workerNode, commonConditions.SucceededType, conditionStatusPointer(metav1.ConditionTrue))
+
+				By("Having no remediation taint and no out-of-service taint")
+				deleteFAR(underTestFAR)
+				Eventually(func(g Gomega) bool {
+					g.Expect(k8sClient.Get(context.Background(), nodeKey, node)).To(Succeed())
+					return utils.TaintExists(node.Spec.Taints, &farNoExecuteTaint) && utils.TaintExists(node.Spec.Taints, &outOfServiceTaint)
```

Comment:
` &&`  should be ` ||`  I guess , otherwise the test succeeds if one taint was removed only

Reply:
Thanks. I will fix this.